src/overmap.cpp

```diff
@@ -1561,81 +1554,35 @@ void overmap::draw(WINDOW *w, WINDOW *wbar, const tripoint &center,
         }
     }
 
-    std::string note_text = overmap_buffer.note(cursx, cursy, z);
-
-    int symbolIndex = note_text.find(':');
-    int colorIndex = note_text.find(';');
-
-    bool symbolFirst = symbolIndex < colorIndex;
-
-    if (colorIndex > -1 && symbolIndex > -1) {
-        if (symbolFirst) {
-            if (colorIndex > 4) {
-                colorIndex = -1;
-            }
-            if (symbolIndex > 1) {
-                symbolIndex = -1;
-                colorIndex = -1;
-            }
-        } else {
-            if (symbolIndex > 4) {
-                symbolIndex = -1;
-            }
-            if (colorIndex > 2) {
-                colorIndex = -1;
-            }
-        }
-    } else if (colorIndex > 2) {
-        colorIndex = -1;
-    } else if (symbolIndex > 1) {
-        symbolIndex = -1;
-    }
-
-    int erasureLength = 0;
+    std::vector<std::string> corner_text;
 
-    if (symbolIndex > -1 && symbolIndex < 5) {
-        if (colorIndex > -1 && !symbolFirst) {
-            erasureLength = symbolIndex;
-        } else if (colorIndex == -1) {
-            erasureLength = symbolIndex;
+    std::string const &note_text = overmap_buffer.note(cursx, cursy, z);
+    if (!note_text.empty()) {
+        size_t const pos = std::get<2>(get_note_display_info(note_text));
+        if (pos != std::string::npos) {
+            corner_text.emplace_back(note_text.substr(pos));
         }
     }
-    if (colorIndex > -1 && colorIndex < 5) {
-        if (symbolIndex > -1 && symbolFirst) {
-            erasureLength = colorIndex;
-        } else if (symbolIndex == -1) {
-            erasureLength = colorIndex;
+   
+    for( const auto &npc : overmap_buffer.get_npcs_near_omt(cursx, cursy, z, 0) ) {
+        if( !npc->marked_for_death ) {
+            corner_text.emplace_back( npc->name );
         }
     }
 
-    if (erasureLength > 0) {
-        note_text.erase(0, erasureLength + 1);
-    }
-
-    std::vector<std::string> corner_text;
-    if (!note_text.empty()) {
-        corner_text.push_back( note_text );
-    }
-    const auto npcs = overmap_buffer.get_npcs_near_omt(cursx, cursy, z, 0);
-    for( const auto &npc : npcs ) {
-        if( npc->marked_for_death ) {
-            continue;
-        }
-        corner_text.push_back( npc->name );
-    }
-    const auto vehicles = overmap_buffer.get_vehicle(cursx, cursy, z);
-    for( const auto &v : vehicles ) {
-        corner_text.push_back( v.name );
+    for( auto &v : overmap_buffer.get_vehicle(cursx, cursy, z) ) {
+        corner_text.emplace_back( std::move(v.name) );
```

Comment:
It's not remotely clear that std::move() is safe here, and even if it is now, it's incredibly fragile, just make a copy.
Also, it's just a string, sheesh.